### DISCUSSION TITLE
Does ccache take -B and -iprefix options into account?

### DISCUSSION THREAD

**Discussion by vijunag** [ccache/ccache#84] [Questions & answers] [answered]
Does ccache take into account -B or -iprefix compiler options during direct or preprocessed mode when CCACHE_BASEDIR is set ? I mean are the paths specified in those switches converted to relative paths ? Does GCC accept relative paths for -B and -iprefix in the first place ? I see quiet a few ccache miss when -B and -iprefix options are included

**Comment by jrosdahl** [comment, marked as answer]
> Does ccache take into account -B or -iprefix compiler options during direct or preprocessed mode when CCACHE_BASEDIR is set ? I mean are the paths specified in those switches converted to relative paths ?
- `-B`: No, not converted to relative path.
- `-iprefix`: Yes, converted to relative path if CCACHE_BASEDIR is set.

> Does GCC accept relative paths for -B and -iprefix in the first place ?
- `-iprefix`: I don't see why it shouldn't. Do you see anything that indicates that it doesn't?
- `-B`: Don't know, but I don't see why it shouldn't.

> I see quiet a few ccache miss when -B and -iprefix options are included

Try enabling ccache logging to get clues on what happens.

Does this answer your questions?

**Comment by jrosdahl** [comment]
I realized after answering that you also submitted #85 (which I had forgotten about since it was a while ago) which hints that the problem you see comes from the concatenated versions of `-B` and `-iprefix` (e.g. `-B/some/path` and `-iprefix/some/path`). I was only considering the separated versions in my reply, e.g. `-iprefix /some/path` which already is rewritten when using a base directory.

**Comment by jrosdahl** [comment]
@vijunag: Could you please try if 5fdb774 solves your problem? (And comments on the approach would also be appreciated, if any.)
